### DISCUSSION TITLE
expose tcp port with k3s scheduler

### DISCUSSION THREAD

**Discussion by finchi** [dokku/dokku#6960] [Q&A]
Hello,

I am currently trying to migrate from the old [dokku-scheduler-kubernetes](https://github.com/dokku/dokku-scheduler-kubernetes) scheduler to the new [k3s scheduler](https://dokku.com/docs/deployment/schedulers/k3s).

In one of my apps, I have a process that needs to be accessible on TCP port 13000. With the old scheduler, I used the [service-process-types](https://dokku.com/docs/deployment/schedulers/k3s) so that the process gets exposed on port 5000, and then used a `ConfigMap` to expose it with `ingress-nginx`.

Is it somehow possible to do this with the new k3s scheduler? How can I tell Dokku to expose another process, besides the web process, to get an open port?

**Comment by josegonzalez** [comment]
Not currently possible, but only because no one has asked for this.

Ideally, what is the workflow you'd want here? Just the same `service-process-types` thing?

**Comment by finchi** [comment]
Ideally, it would be great if I didn't have to touch k3s at all for this. Ideally, I would like to configure everything solely with dokku commands.

dokku `ports:add` comes to mind, but as far as I understand, there is currently no way to target other processes except the web process.

However, `service-process-types` would be totally fine, since I don't think many people have such a use case.

**Reply by josegonzalez** [reply]
Mind filing a ticket for this? We can probably easily do the `service-process-types` hack and then come back and figure out a proper solution long-term.

**Reply by finchi** [reply]
Sorry for coming back with this so late. Created a feature request #7204

Thank you
